<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="gzpose" default="-x 0.0 -y 0.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/>

    <include file="$(find tiago_2dnav_gazebo)/launch/tiago_navigation.launch">
        <arg name="public_sim"   value="true"/>
        <arg name="world"        value="prenovelty_domain"/> 
        <arg name="gzpose"       value="$(arg gzpose)"/>
        <arg name="end_effector" value="pal-gripper"/>
    </include>

    <node pkg="robot_custom" type="get_object_pose.py" name="get_object_pose" output="screen"/>
    <node pkg="robot_custom" type="object_nav.py" name="object_nav" output="screen"/>
    <node pkg="robot_custom" type="waypoint_nav.py" name="waypoint_nav" output="screen"/>
    <node pkg="robot_custom" type="pose_nav.py" name="pose_nav" output="screen"/>
    <node pkg="robot_custom" type="map_to_gazebo_broadcaster.py" name="map_to_gazebo_broadcaster" output="screen"/>
    <node pkg="robot_custom" type="at_publisher.py" name="at_publisher" output="screen"/>
    <node pkg="robot_custom" type="facing_publisher.py" name="facing_publisher" output="screen"/>

    <!-- marker detector -->
    <node pkg="aruco_ros" type="single"  name="aruco_single">
        <remap from="/camera_info"       to="/xtion/rgb/camera_info" />
        <remap from="/image"             to="/xtion/rgb/image_rect_color" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="0.045"/>
        <param name="marker_id"          value="582"/>
        <param name="reference_frame"    value="base_footprint"/>   <!-- frame in which the marker pose will be expressed -->
        <param name="camera_frame"       value="xtion_rgb_optical_frame"/>
        <param name="marker_frame"       value="aruco_frame" />
        <param name="corner_refinement"  value="SUBPIX" />
    </node>

    <!-- Load definition of pregrasp motion into play_motion namespace -->
    <rosparam command="load" file="$(find tiago_pick_demo)/config/pick_motions.yaml" />

    <!-- Pick & place server -->
    <node name="pick_and_place_server" pkg="tiago_pick_demo" type="pick_and_place_server.py" output="screen">
        <rosparam command="load" file="$(find tiago_pick_demo)/config/pick_and_place_params.yaml" />
        <param name="object_width"  value="0.055" />
        <param name="object_height" value="0.1" />
        <param name="object_depth"  value="0.055" />
    </node>

    <rosparam command="load" file="$(find robot_custom)/config/objects.yaml"/>
    <rosparam command="load" file="$(find robot_custom)/config/waypoints.yaml"/>
    <rosparam command="load" file="$(find robot_custom)/config/transforms.yaml"/>
    <rosparam command="load" file="$(find robot_custom)/config/boundaries.yaml"/>
</launch>